Code/Framework/AzCore/AzCore/Name/Name.h

```diff
@@ -82,62 +114,101 @@ namespace AZ
             return GetHash() != other.GetHash();
         }
 
-        // We delete these operators because using Name in ordered containers is not supported. 
+        bool operator==(const NameRef& other) const
+        {
+            return m_data == other;
+        }
+
+        bool operator!=(const NameRef& other) const
+        {
+            return m_data != other;
+        }
```

Comment:
These operators aren't actually needed since there is an operator NameRef that would convert a Name to the NameRef before performing the comparison

Reply:
The global operators might actually help, I'll do some tests - when I ported over the Generic DOM as a test I did see a case or two where we could have ambiguity in `operator==` because it wasn't sure if it should promote both sides to `Name` or `NameRef`

Reply:
I found a few cases in which operator overload could be ambiguous because of `intrusive_ptr`'s built in equality operators - I built a tiny shim around the data for `NameRef` instead, which works just fine in tests - open to feedback about how we feel about this approach (tests + DOM stuff works with the new NameRef)